local_pathfinding/objectives.py

```diff
@@ -24,66 +53,313 @@ def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        """
+
+        # Calculate the angle from s1-goal line segment from North
+        global_goal_direction = math.atan2(self.goal_x - s1.getX(), self.goal_y - s1.getY())
+
+        angle = self.min_turn_angle(global_goal_direction, self.heading)
+        return ob.Cost(angle)
+
+    def heading_path_turn_cost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
+        """Generates the turning cost between s1-s2 and heading of the sailbot
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            ob.Cost: The minimum turning angle between s1-s2 and heading  (degrees)
+        """
+
+        # Calculate the angle of the s1-s2 line segment from North
+        path_direction = math.atan2(s2.getX() - s1.getX(), s2.getY() - s1.getY())
+
+        angle = self.min_turn_angle(path_direction, self.heading)
+        return ob.Cost(angle)
+
+    @staticmethod
+    def min_turn_angle(angle1: float, angle2: float) -> float:
+        """Calculates the minimum turning angle between two angles
+
+        Args:
+            angle1 (float): The first angle in radians
+            angle2 (float): The second angle in radians
+
+        Returns:
+            float: The minimum turning angle between the two angles (degrees)
+        """
+
+        # Calculate the uncorrected turn size [0, 2pi]
+        turn_size_bias = math.fabs(angle1 - angle2)
+
+        # Correct the angle in between [0, pi]
+        if turn_size_bias > math.pi:
+            turn_size_unbias = 2 * math.pi - turn_size_bias
+        else:
+            turn_size_unbias = turn_size_bias
+
+        return math.degrees(math.fabs(turn_size_unbias))
 
 
 class WindObjective(Objective):
-    def __init__(self, space_information, wind_direction_degrees: float):
+    """Generates a wind objective function
+
+    Attributes:
+        wind_direction (float): The direction of the wind in radians [-pi, pi)
+    """
+
+    def __init__(self, space_information, wind_direction: float):
         super().__init__(space_information)
-        self.wind_direction_degrees = wind_direction_degrees
+        self.wind_direction = math.radians(wind_direction)
 
     # This objective function punishes the boat for going up/downwind
     def motionCost(self, s1: ob.SE2StateSpace, s2: ob.SE2StateSpace) -> ob.Cost:
-        return ob.Cost()
+        """Generates the cost associated with the upwind and downwind directions of the boat in
+           relation to the wind
+
+        Args:
+            s1 (SE2StateInternal): The starting point of the local start state
+            s2 (SE2StateInternal): The ending point of the local goal state
+
+        Returns:
+            float: The cost of going upwind or downwind
+        """
+        distance = ((s2.getY() - s1.getY()) ** 2 + (s2.getX() - s1.getX()) ** 2) ** 0.5
```

Comment:
I saw you were using a hypotenuse function elsewhere, should that maybe be used here?

Reply:
This works though, its fine 👍